core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -276,7 +276,7 @@ final class ShuffleBlockFetcherIterator(
           val (size, mapIndex) = infoMap(blockId)
           FetchBlockInfo(BlockId(blockId), size, mapIndex)
         }
-        results.put(DeferFetchRequestResult(FetchRequest(address, blocks.toSeq)))
+        results.put(DeferFetchRequestResult(FetchRequest(address, blocks.toIndexedSeq)))
```

Comment:
When using Scala 2.12, `blocks.toSeq` is a redundant operation, which will not cause any memory copy . However, 'blocks.toIndexedSeq' will cause a memory copy. This change may be beneficial to Scala 2.13, but it will harm the performance of Scala 2.12. 

The current default version is still Scala 2.12, so I suggest leave this line as it is and add some TODOs to facilitate change when Scala 2.13 is the default version

Reply:
Maybe defining a helper method and providing different implementations for Scala 2.12 and 2.13 can give consideration to both, but this may be too complicated.